### PR TITLE
8324342: Doclet should default @since for a nested class to that of its enclosing class

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review changes to the support for `since`, `author`, and `version` tags, such that if there are no such tags on a nested class, the doclet will look for any such tags in enclosing classes.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Change requires CSR request [JDK-8324735](https://bugs.openjdk.org/browse/JDK-8324735) to be approved
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8324342](https://bugs.openjdk.org/browse/JDK-8324342): Doclet should default @since for a nested class to that of its enclosing class (**Enhancement** - P3)
 * [JDK-8324735](https://bugs.openjdk.org/browse/JDK-8324735): Doclet should default @since for a nested class to that of its enclosing class (**CSR**)

### Reviewers
 * [Pavel Rappo](https://openjdk.org/census#prappo) (@pavelrappo - **Reviewer**)
 * [Hannes Wallnöfer](https://openjdk.org/census#hannesw) (@hns - **Reviewer**) ⚠️ Review applies to [7a1dc75c](https://git.openjdk.org/jdk/pull/18407/files/7a1dc75ced2de42b52d8db1939d80bcea74800be)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18407/head:pull/18407` \
`$ git checkout pull/18407`

Update a local copy of the PR: \
`$ git checkout pull/18407` \
`$ git pull https://git.openjdk.org/jdk.git pull/18407/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18407`

View PR using the GUI difftool: \
`$ git pr show -t 18407`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18407.diff">https://git.openjdk.org/jdk/pull/18407.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18407#issuecomment-2010401963)

### CODE REVIEW

**Review by pavelrappo — COMMENTED** [review]
This looks like the right thing to do; thanks for doing this.

A general comment regarding the test cases. It would be better if:

 - the nesting was at least two levels deep (`public class Foo { static class Bar { static class Baz { } } }`), to test search recursion
 - some tags were repeated (IIRC, `@since` and `@autor` allow for repetition), to test that all such tags from the enclosing context are used

**Review by hns — APPROVED** [review]
Looks good. I'll leave it to you and @pavelrappo to decide whether to include the partially unrelated updates to the man page.

File: src/jdk.javadoc/share/man/javadoc.1 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.\" Copyright (c) 1994, 2023, Oracle and/or its affiliates. All rights reserved.
+.\" Copyright (c) 1994, 2024, Oracle and/or its affiliates. All rights reserved.
```
[the remark this answers is not in the document]
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/man/javadoc.1:1]: The changes were all described together in the associated CSR, which you reviewed.

[the remark this answers is not in the document]
Comment (pavelrappo) [reply src/jdk.javadoc/share/man/javadoc.1:1]: Hm... I cannot find where, for example, these hunks were described:

```
 For more details on the conformance requirements for HTML5 documents,
-see \f[B]Conformance requirements\f[R]
-[https://www.w3.org/TR/html5/infrastructure.html#conformance-requirements]
+see \f[B]Conformance requirements for authors\f[R]
+[https://html.spec.whatwg.org/multipage/introduction.html#conformance-requirements-for-authors]
 in the HTML5 Specification.
```

```
-This command inserts the following line in the head of every generated
-page:
+This command inserts the following line, containing a
+\f[B]\f[VB]meta\f[B] element\f[R]
+[https://html.spec.whatwg.org/multipage/semantics.html#the-meta-element]
+in the head of every generated page:
```

**Comment by jonathan-gibbons** [comment]
> This looks like the right thing to do; thanks for doing this.
> 
> A general comment regarding the test cases. It would be better if:
> 
> * the nesting was at least two levels deep (`public class Foo { static class Bar { static class Baz { } } }`), to test search recursion
> * some tags were repeated (IIRC, `@autor` allows for repetition), to test that all such tags from the enclosing context are used

Extended tests for double-nested classes and multiple tags

**Review by pavelrappo — APPROVED** [review]
Thanks for adding extra tests. Looks good.

**Pull request closed without merging**